### PR TITLE
fix(cdn,ui): fix download run-result + clean buffer

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/cdn/storage/redis/redis.go [hunk 1]
```diff
@@ -27,8 +27,14 @@ type Redis struct {
 	bufferType storage.CDNBufferType
 }
 
+const driverName = "redis"
```
Comment (richardlt) [inline engine/cdn/storage/redis/redis.go:30]: Can be add a string alias type in SDK with all values at the same place ?
```
type CDNStorageUnitType string
const (
CDNUnitRedis CDNStorageUnitType = "redis"
...
)
```

**Review by fsamin — APPROVED** [review]

File: engine/cdn/storage/redis/redis.go (hunk 1 above)
Comment (fsamin) [reply engine/cdn/storage/redis/redis.go:30]: humm, do you think it's really important to have a dependency between the storage subpackage and the whole sdk for just one usage ?

Comment (sguiheux) [reply engine/cdn/storage/redis/redis.go:30]: i like the idea that a storage bring his driver

File: engine/cdn/storage/types.go [hunk 2]
```diff
@@ -284,6 +285,27 @@ func (x RunningStorageUnits) GetBuffer(bufferType sdk.CDNItemType) BufferUnit {
...
+		for _, unit := range x.Storages {
+			if unit.GetDriverName() == "cds" {
+				cdsBackendID = unit.ID()
+				break
+			}
+		}
+
+		for i, s := range ius {
+			if s.UnitID == cdsBackendID {
+				ius = append(ius[:i], ius[i+1:]...)
```
Comment (richardlt) [inline engine/cdn/storage/types.go:301]: Not sure that it works if i == len(ius)-1

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15756.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

File: engine/cdn/storage/types.go (hunk 2 above)
Comment (sguiheux) [reply engine/cdn/storage/types.go:301]: it works

**Review by yesnault — APPROVED** [review]

**Review by richardlt — APPROVED** [review]

**Pull request merged**